src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowsCache.cs

```diff
@@ -66,8 +74,47 @@ public bool TryGetValue(string key, out SKImage? image)
 
 	public bool Remove(string key)
 	{
-		System.Diagnostics.Debug.WriteLine($"[ShadowsCache] Remove => key: {key}");
-
 		return _shadowsCache.TryRemove(key, out var _);
 	}
+	public async void CleanupIfNeeded()
+	{
+		if ((DateTime.UtcNow - _lastCleanupTime).TotalSeconds > CleanupInterval)
+		{
+			try
+			{
+				await CleanupAsync();
+			}
+			catch (Exception ex)
+			{
+				System.Diagnostics.Debug.WriteLine($"[ShadowsCache] Cleanup failed: {ex}");
+			}
+		}
+	}
+
+	/// <summary>
+	/// Removing old shadows based on their stats.
+	/// </summary>
+	private Task CleanupAsync()
+	{
+		_lastCleanupTime = DateTime.UtcNow;
+
+		return Task.Run(() =>
+		{
+			Stopwatch stopwatch = new Stopwatch();
+			stopwatch.Start();
+			System.Diagnostics.Debug.WriteLine($"[ShadowsCache] Cleanup starting");
+			int removedShadows = 0;
+			foreach (var expiredBucket in _shadowsCache
+				.Where(x => OneHitSinceAShortTime(x.Value) || SeveralHitsSinceALongTime(x.Value)))
```

Comment:
you can inline the where clause inside foreach's bracket

Reply:
and once you inline them, you can reuse `_lastCleanupTime` for comparisons instead of re-retrieving `.UtcNow` again

Reply:
I would rather not inline those cause the readiness would be significantly worse for a insignificant performance improvement. We are already in a new Task, likely a different thread, and we don't need those microseconds here.
For the same reason we can't reuse _lastCleanupTime cause it will likely cause corruption (assigned and read from differents threads).